sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterPane.java

```diff
@@ -100,6 +100,11 @@ public void clear(TriggerContext c) throws Exception {
   }
 
   @Override
+  public boolean hasState(TriggerContext c) throws Exception {
+    return !c.state().access(ELEMENTS_IN_PANE_TAG).isEmpty().read();
```

Comment:
Needs tests. In particular, the situations where you expect no state vary per-trigger.

There already were some tests like that, but I may have lost them in the trigger shuffle. Your new public method would make them more elegant than previously, when they relied on `TriggerTester` scraping the state table.